Modules/_xxsubinterpretersmodule.c

```diff
@@ -263,7 +263,9 @@ interp_create(PyObject *self, PyObject *args, PyObject *kwds)
         return NULL;
     }
 
+    PyThreadState_Swap(tstate);
```

Comment:
The `PyInterpreterState_GetIDObject()` call needs to happen with `save_tstate` active because it creates a Python object (a `InterpreterID`) that will be returned in the `save_tstate` context.